integration/compose/src/main/java/com/bumptech/glide/integration/compose/GlidePainter.kt

```diff
@@ -36,54 +45,345 @@ import kotlinx.coroutines.SupervisorJob
+): Pair<Painter, GlidePainterState> {
+  val requestManager: RequestManager = LocalContext.current.let { remember(it) { Glide.with(it) } }
+
+  val requestBuilder = remember(model, requestManager, requestBuilderTransform) {
+    requestBuilderTransform(requestManager.load(model))
+  }
+
+  val overrideSize = requestBuilder.overrideSize()
+  val size = rememberResolvableSize(overrideSize)
+  val painter = rememberGlidePainter(
+    requestBuilder = requestBuilder,
+    size = size,
+    resolveSize = true,
+  )
+  return Pair(painter, painter.state)
+}
+
+@ExperimentalGlideComposeApi
+public data class GlidePainterStateAndModifier(
+  val painter: Painter,
+  val state: GlidePainterState,
+  val modifier: Modifier,
+)
+
+/**
+ * Similar to the other variant of [rememberGlidePainter] except that this method uses the given
+ * [modifier] to determine the size for the image load. See the documentation on the other variant
+ * for details.
+ *
+ * This method returns an updated [GlidePainterStateAndModifier.modifier] which must be applied to
+ * some other Composable whose layout size is the size of the image you want. For example, using
+ * this method and the returned `modifier` allows you avoid having to specify an `override()` size:
+ *
+ * ```
+ * val (painter, state, updatedModifier) = rememberGlidePainter(model = item.uri, modifier)
+ * when (state.animate()) {
+ *  AnimationState.Loading, AnimationState.Animate -> {
+ *    Crossfade(state == AnimationState.Loading) {
+ *      if (it) {
+ *        Image(
+ *          painterResource(android.R.drawable.star_big_on),
+ *          contentDescription,
+ *          updatedModifier,
+ *        )
+ *      } else {
+ *        Image(painter, contentDescription, updatedModifier)
+ *      }
+ *    }
+ *  }
+ *  AnimationState.Success -> {
+ *    Image(painter, contentDescription, updatedModifier)
+ *  }
+ *  AnimationState.Failed -> {
+ *    Image(
+ *      painterResource(android.R.drawable.star_big_off),
+ *      contentDescription,
+ *      updatedModifier,
+ *    )
+ *  }
+ * }
+ * ```
+ *
+ * If you're using fixed size buckets across the app, or have some other reason to use
+ * [RequestBuilder.override], then this method is not helpful and you should use the other
+ * [rememberGlidePainter] variant instead. If an override size is set, this method will throw.
+ *
+ * Otherwise, using this method can improve efficiency by ensuring you load an image whose size
+ * matches the layout it's displayed in. In addition, this variant will always resolve a size and
+ * be able to start an image load if the the returned [GlidePainterStateAndModifier.modifier] is
+ * used and has a reasonable layout size.
+ *
+ * @throws IllegalArgumentException if [requestBuilderTransform] sets an [RequestBuilder.override]
+ * size.
+ */
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+@Composable
+public fun rememberGlidePainter(
+  model: Any?,
+  modifier: Modifier,
```

Comment:
Typically, modifiers are used for UI elements and not in `remember` functions. I don't have a good suggestion here though on how to accomplish what you are trying to do however as I understand that the modifier is the mechanism for determining the image size.

Reply:
One other option is extracting a method that deals with modifiers and then having people pass in ResolvableGlideSize to rememberGlidePainter instead:

```kotlin
  val (size, updatedModifier) = waitForSizeFromModifier(modifier);
  val painter = rememberGlidePainter(model, size);
  Image(painter, contentDescription, updatedModifier)
```

Maybe that's a bit nicer?

Reply:
Done